pyresample/_spatial_mp.py

```diff
@@ -106,12 +111,26 @@ def query(self, x, k=1, eps=0, p=2, distance_upper_bound=np.inf):
 
 
 class BaseProj(pyproj.Proj):
+
+    def __init__(self, projparams=None, preserve_units=True, **kwargs):
+        if pyproj.__version__ >= '2':
+            warn_msg = '{"init": "EPSG:XXXX"} is no longer supported. Use "EPSG:XXXX" as a proj string instead'
+            if 'init' in kwargs:
+                warnings.warn('init="EPSG:XXXX" is no longer supported. Use "EPSG:XXXX" as a proj string instead')
+                projparams = int(kwargs.pop('init').replace('EPSG:', ''))
+            elif isinstance(projparams, Iterable) and 'init' in projparams:
+                # Dicts are cleaner to parse than strings.
+                if isinstance(projparams, str):
+                    warn_msg = '"+init=EPSG:XXXX" is no longer supported. Use "EPSG:XXXX" as a proj string instead'
+                    projparams = proj4_str_to_dict(projparams)
+                warnings.warn(warn_msg)
+                projparams = int(projparams.pop('init').replace('EPSG:', ''))
```

Comment:
I think the preferred/supported way going forward with pyproj is to do `"EPSG:4326"` as a string.